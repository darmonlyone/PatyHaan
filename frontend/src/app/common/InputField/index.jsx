import React from "react";

const InputField = ({ testid, title, type, className, name, onChange, value }) => {
	return (
		<div className="my-2 w-full">
			<span className="mb-2 text-sm">{title}</span>
			<input
				data-testid={testid}
				value={value}
				name={name}
				onChange={onChange}
				className={className || "text-md w-full"}
				type={type || "text"}
				required
			/>
		</div>
	);
};

export default InputField;
