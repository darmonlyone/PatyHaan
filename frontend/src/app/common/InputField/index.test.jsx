import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import InputField from ".";

let container = null;
beforeEach(() => {
	container = document.createElement("div");
	document.body.appendChild(container);
});

afterEach(() => {
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

it("should renders InputField", () => {
	act(() => {
		render(<InputField />, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

    act(() => {
		render(<InputField title={"test"}/>, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

    act(() => {
		render(<InputField title={"test2"} className="w-full bg-gray-100" />, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

    act(() => {
		render(<InputField title={"test2"} type={"password"} />, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

});
