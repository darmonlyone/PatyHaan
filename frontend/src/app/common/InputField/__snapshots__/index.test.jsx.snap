// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders InputField 1`] = `"<div class=\\"my-2 w-full\\"><span class=\\"mb-2 text-sm\\"></span><input class=\\"text-md w-full\\" type=\\"text\\" required=\\"\\" value=\\"\\"></div>"`;

exports[`should renders InputField 2`] = `"<div class=\\"my-2 w-full\\"><span class=\\"mb-2 text-sm\\">test</span><input class=\\"text-md w-full\\" type=\\"text\\" required=\\"\\" value=\\"\\"></div>"`;

exports[`should renders InputField 3`] = `"<div class=\\"my-2 w-full\\"><span class=\\"mb-2 text-sm\\">test2</span><input class=\\"w-full bg-gray-100\\" type=\\"text\\" required=\\"\\" value=\\"\\"></div>"`;

exports[`should renders InputField 4`] = `"<div class=\\"my-2 w-full\\"><span class=\\"mb-2 text-sm\\">test2</span><input class=\\"text-md w-full\\" type=\\"password\\" required=\\"\\" value=\\"\\"></div>"`;
